### ISSUE TITLE
Start with docker swarm got error connecting to ws://minio2:9000/minio/grid/v1: lookup minio2 on 127.0.0.11:53: no such host

### ISSUE THREAD

**Issue by seepine** [minio/minio#19963]
### Discussed in https://github.com/minio/minio/discussions/19496

<div type='discussions-op-text'>

<sup>Originally posted by **seepine** April 13, 2024</sup>
Use docker compose yml https://github.com/minio/minio/blob/master/docs/orchestration/docker-compose/docker-compose.yaml

```
Error: grid: http://minio1:9000 connecting to ws://minio2:9000/minio/grid/v1: lookup minio2 on 127.0.0.11:53: no such host (*net.DNSError) Sleeping 1.065s (3) (*fmt.wrapError)
       6: internal/logger/logonce.go:118:logger.(*logOnceType).logOnceIf()
       5: internal/logger/logonce.go:149:logger.LogOnceIf()
       4: internal/grid/connection.go:59:grid.gridLogOnceIf()
       3: internal/grid/connection.go:682:grid.(*Connection).connect.func1()
       2: internal/grid/connection.go:688:grid.(*Connection).connect()
       1: internal/grid/connection.go:260:grid.newConnection.func1()
ERROR Unable to start the server: context canceled
```</div>

**Comment by seepine** [comment]
#19610

**Comment by andreav** [comment]
I read both issues (this and #19610) but I did not understand if the issue has been resolved.
I'm having the exact same problem right now.

Other containers attached in the same network work as expected.
minio containers do not resolve their addresses using their service names. 

It is strage because i can nslookup "minio", and also "tasks.minio" but not my real services "minio1" and "minio2":

```
bash-5.1# nslookup tasks.minio
Server:         127.0.0.11
Address:        127.0.0.11#53

Non-authoritative answer:
Name:   tasks.minio
Address: 10.0.32.7
Name:   tasks.minio
Address: 10.0.32.6
```

But nslookup should resolve when invoked with "minio1" and "minio2", and this does not happen.

I tried also using "hostname" and "network aliases" inside the compose.yml but the result is always the same.
